### ISSUE TITLE
Redesign

### ISSUE THREAD

**Issue by amilajack** [amilajack/popcorn-time-desktop#145]
An initial base design for release 0.0.6

<img width="784" alt="screen shot 2016-08-19 at 9 39 25 am" src="https://cloud.githubusercontent.com/assets/6374832/17841426/32e35194-67cd-11e6-84ab-dc5c82e49578.png">

**Comment by ZakariaRidouh** [comment]
I love how the Popcorn Icon is in the middle. Also, I like how there would (is there?) be a large cover picture of trending movies in the forefront. Do you have a color pallet already made?

💯

**Comment by vzamanillo** [comment]
Looks very cool!

**Comment by amilajack** [comment]
@ZakariaRidouh The biggest gray rectangle is for that.

**Comment by amilajack** [comment]
@vzamanillo That makes sense. I'll make an updated version today with the changes.

**Comment by vzamanillo** [comment]
Something like this, maybe?

<img width="784" alt="32e35194-67cd-11e6-84ab-dc5c82e49578" src="https://cloud.githubusercontent.com/assets/10209695/17885232/9adb0aa8-691c-11e6-8ceb-0f9b528f5eb5.png">

**Comment by amilajack** [comment]
I was thinking of something like this:

![Mockup](https://github.com/amilajack/popcorn-time-desktop-design/raw/master/Desktop.jpg)

What do you think?

**Comment by vzamanillo** [comment]
I think that looks really cool! love it, but, how to select the featured movies? A carousel for 'Popular' and 'Recently Watched' movies (with a limited number of movies by a fixed default value or a configurable value) could be great too. Maybe the recently watched movies could be retieved from trakt.tv or tvshowtime, but needs scrobbling... Where you are planning to place the movie categories? What information should the hover design show? great work :)

**Comment by amilajack** [comment]
> Maybe the recently watched movies could be retieved from trakt.tv or tvshowtime, but needs scrobbling

This is extra work. I was thinking of simply using a local database or storage for this. When the user clicks a movie, save its imdb to a list.

> Where you are planning to place the movie categories?

Movie categories are going to be placed in the 'Movies' page. The featured page simply shows what's trending, recommendations and personalized content. 'Movies' and 'Shows' will be similar to the existing movies and shows pages in the current app. In those pages, users will be able to filter movies by category and change the category of movies.

> What information should the hover design show?

Good question. I have no idea. I'll need to think it through. There's lots of information I could show but I dont want to clutter the UI too much. Here's a list of components I'm thinking of showing on hover inside the Movie component:

* Progress of movie: How much percentage of the movie the user has watched
* Genre
* Title
* Rating
* Year

Idea:

<img width="880" alt="screen shot 2016-08-23 at 2 00 49 pm" src="https://cloud.githubusercontent.com/assets/6374832/17909634/2b152e6a-693a-11e6-8a03-5c0417e03d38.png">

**Comment by vzamanillo** [comment]
> This is extra work. I was thinking of simply using a local database or storage for this. When the user clicks a movie, save its imdb to a list.

For 'Recently watched' the movie should be saved when the user view the complete movie, if the user click to watch a movie and view one minute only (imagine, the film is so bad) the movie should be on a 'pending' state, a possibility is bring to the user an option to remove the pending movies only, I think adding another section to pending movies has no sense. An adapter to save to db, trakt.tv or tvshowtime will be optimal.

What do you think about adding an 'Upcoming movies section'? some API like [tmdb](https://www.themoviedb.org/) offers [this feature](https://www.themoviedb.org/movie/upcoming)

> Movie categories are going to be placed in the 'Movies' page. The featured page simply shows what's trending, recommendations and personalized content. 'Movies' and 'Shows' will be similar to the existing movies and shows pages in the current app. In those pages, users will be able to filter movies by category and change the category of movies.

Based on the categories of the recently watched movies, rating... etc, sounds good.

> Good question. I have no idea. I'll need to think it through. There's lots of information I could show but I dont want to clutter the UI too much. Here's a list of components I'm thinking of showing on hover inside the Movie component:

'Director' and 'Awards', but 'Awards' could disturb a bit.

> Also one of my goals with this project is to keep communication as open as possible. I really want to encourage open discussion about how we can improve this client. So, in my opinion, there's no such thing as too many questions.

Completely agree, good philosophy, that's how great things are done.

**Comment by amilajack** [comment]
> An adapter to save to db, trakt.tv or tvshowtime will be optimal.

I've looked into this before and its actually a lot of work. For that, we'll need to manage authentication, storage of credentials, and other information. If users want this functionality, I can expose the underlying API so that I can be extended to devs, who can write their own plugins for trakt and other providers.

Here's what I think the basic functionality of playback position capture should be like:
1. Detect the player (PCT, vlc, chromecast, etc) and retrieve the current playback
2. Save the current playback position in minutes along with the movie's IMDB id. This information will be persisted to a local database or cache.
3. On playback of a movie, check the database to see if the movie was played previously. if it was, retrieve the progress of the movie (where the user left off) and start the movie from there.

> What do you think about adding an 'Upcoming movies section'?

Adding new releases is actually pretty difficult. We need to guarantee that our providers can actually supply the movie source of these new movies, and thats usually very unlikely.

> 'Director' and 'Awards', but 'Awards' could disturb a bit.

This functionality will likely be added to the Movie (not movies) page. In my opinion, adding it to the movie card is a bit too much clutter. We can always prototype this out to check if that's the case.

> Completely agree, good philosophy, that's how great things are done.

Couldn't agree more!

**Comment by amilajack** [comment]
**Updated movies page:**

![popcorn-time-desktop-movie](https://cloud.githubusercontent.com/assets/6374832/18035784/459249b4-6d11-11e6-810d-8f9cc4b95b70.gif)

**Comment by vzamanillo** [comment]
Awesome, easy categories access and navigation.

**Comment by amilajack** [comment]
Check out the live demo for the animation:

```bash
git clone https://github.com/amilajack/animation-experiments.git
cd animation-experiments
npm i
open staggered-cards/index.html
```

**Comment by amilajack** [comment]
Also the source for the designs is in this repo:

https://github.com/amilajack/popcorn-time-desktop-design

**Comment by amilajack** [comment]
![animation](https://github.com/amilajack/popcorn-time-desktop-design/blob/master/popcorn-time-desktop-animation.gif?raw=true)

**Comment by onbjerg** [comment]
@amilajack Very nice. Can we still have control over the video quality somewhere, though?

**Comment by amilajack** [comment]
Yes. It isnt shown in the demo but it would look something like this:

<img width="292" alt="screen shot 2016-10-16 at 3 57 37 pm" src="https://cloud.githubusercontent.com/assets/6374832/19421580/4e3eecce-93b9-11e6-83a9-c66204972a6d.png">

Not that these designs are subject to change.

**Comment by aurnik** [comment]
For the featured page, would the jumbotron/spotlight area be a slider? It seems like it would be a good place to showcase a few different movies in a way that looks slightly different from the individual movie page.
And for hover information I wanted to add: is there currently anything in place to track the viewcount/popularity of an item? If so that would be cool to see.

(Also just wanted to introduce myself real quick, just found the project through the official PT github trying to look for an Electron version of the project. Hope I can be of help both design and development -wise.)

**Comment by aurnik** [comment]
@amilajack Is this still an active project?

**Comment by amilajack** [comment]
I've been having a bunch of finals in school so I haven't had much time to contribute to this project or any other project that I contribute to 😔 Right now, my time is divided between [eslint-plugin-flowtype-errors](https://github.com/amilajack/eslint-plugin-flowtype-errors), [eslint-plugin-compat](https://github.com/amilajack/eslint-plugin-compat), and mostly [electron-react-boilerplate](https://github.com/chentsulin/electron-react-boilerplate).

But over winter break, I will very likely make a significant number of contributions to this project. Also this project could really benefit from some contributors. If you're interested in being a contributor, let me know.

**Comment by aurnik** [comment]
I'd love to contribute -- I was sad to see that this project was dead on the official repo but happy to find that you were reviving it.

Do you have any data on the performance improvements between this and the currently used PC app? It could help to bring in new contributors.

**Comment by amilajack** [comment]
Contrary to popular belief, this project is not a fork of any popcorn time or butter project. Its actually a fork of [electron-react-boilerplate](https://github.com/chentsulin/electron-react-boilerplate). So its not related to the official repo.

I've been looking for contributor for a while but couldn't seem to get any. I think two people is actually enough to bring this project back to life. Lets discuss this in #617.

**Comment by amilajack** [comment]
@aurnik I've invited you to the gitter chat.

**Comment by aurnik** [comment]
@amilajack For this project or for electron-react-boilerplate?

**Comment by CHEF-KOCH** [comment]
Another design thing is that in full screen you still see the menu bar. 

![wdd](https://user-images.githubusercontent.com/727123/28375703-7703ada2-6ca8-11e7-823b-2353568b5976.png)

**Comment by amilajack** [comment]
Maybe we can have a logo like this:
https://codepen.io/amilajack/full/xXWRmo/
